Philips.CodeAnalysis.Test/Verifiers/DiagnosticVerifier.cs

```diff
@@ -47,9 +47,30 @@ protected void VerifyDiagnostic(string source, DiagnosticId id)
 				Message = new Regex(".*"),
 				Severity = DiagnosticSeverity.Error,
 			};
-			VerifyDiagnostic(source, diagnosticResult);
+			var analyzer = GetDiagnosticAnalyzer();
+			VerifyDiagnosticsInternal(new[] { source }, filenamePrefix, assemblyName, analyzer, new[] { diagnosticResult });
 		}
 
+		protected void VerifyDiagnostic(string source, int count)
+		{
+			Assert.IsTrue(count > 1, "Only use this overload when your test expects the same Diagnostic multiple times.");
+			var analyzer = GetDiagnosticAnalyzer() as SingleDiagnosticAnalyzer;
+			Assert.IsNotNull(analyzer, @"This overload is only supported for Analyzers that support a single DiagnosticId");
```

Comment:
We could create a `SingleDiagnosticVerifier` class, that only supports the `SingleDiagnosticAnalyzer` and eliminate this check.

Reply:
Agreed. Filed Issue.